draft-ietf-quic-tls.md

```diff
@@ -1007,144 +813,94 @@ encrypted_pn = ChaCha20(pn_key, counter, nonce, packet_number)
+before the final TLS handshake messages are received.  A client will be unable
+to decrypt 1-RTT packets from the server, whereas a server will be able to
+decrypt 1-RTT packets from the client.
 
-A HelloRetryRequest handshake message might be used to reject an initial
-ClientHello.  A HelloRetryRequest handshake message is sent in a Retry packet;
-any second ClientHello that is sent in response uses a Initial packet type.
-These packets are only protected with a predictable key (see
-{{handshake-secrets}}).  This is natural, because no shared secret will be
-available when these messages need to be sent.  Upon receipt of a
-HelloRetryRequest, a client SHOULD cease any transmission of 0-RTT data; 0-RTT
-data will only be discarded by any server that sends a HelloRetryRequest.
+Packets protected with 1-RTT keys MAY be stored and later decrypted and used
+once the handshake is complete.  A server MUST NOT use 1-RTT protected packets
+before verifying either the client Finished message or - in the case that the
+server has chosen to use a pre-shared key - the pre-shared key binder (see
+Section 4.2.8 of {{!TLS13}}).  Verifying these values provides the server with
+an assurance that the ClientHello has not been modified.
 
-The packet type ensures that protected packets are clearly distinguished from
-unprotected packets.  Loss or reordering might cause unprotected packets to
-arrive once 1-RTT keys are in use, unprotected packets are easily distinguished
-from 1-RTT packets using the packet type.
+A server could receive packets protected with 0-RTT keys prior to receiving a
+TLS ClientHello.  The server MAY retain these packets for later decryption in
+anticipation of receiving a ClientHello.
 
-Once 1-RTT keys are available to an endpoint, it no longer needs the TLS
-handshake messages that are carried in unprotected packets.  However, a server
-might need to retransmit its TLS handshake messages in response to receiving an
-unprotected packet that contains ACK frames.  A server MUST process ACK frames
-in unprotected packets until the TLS handshake is reported as complete, or it
-receives an ACK frame in a protected packet that acknowledges all of its
-handshake messages.
+Receiving and verifying the TLS Finished message is critical in
+ensuring the integrity of the TLS handshake.  A server MUST NOT use
+1-RTT protected packets from the client prior to verifying the client
+Finished message if its response depends on client authentication.
 
-To limit the number of key phases that could be active, an endpoint MUST NOT
-initiate a key update while there are any unacknowledged handshake messages, see
-{{key-update}}.
 
+# Key Update
 
-## Key Update {#key-update}
+Once the 1-RTT keys are established and the short header is in use, it
+is possible to update the keys, for instance because of limits on AEAD
+encryption.  The KEY_PHASE bit in the short header is used to indicate
+whether key updates have occurred. The KEY_PHASE bit (the 0x20 bit of
```

Comment:
0x40. I don't think we need to reference the exact position of the bit here at all.

Reply:
Done